### PR TITLE
Alin

### PR DESCRIPTION
**Pull request by alinulms**
Alin - Initial commit.

### CODE REVIEW

File: src/Sitecore.Pathfinder.Console/Building/Checking/CheckSolution.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using System.Threading.Tasks;
+using Sitecore.Pathfinder.Diagnostics;
+
+namespace Sitecore.Pathfinder.Building.Checking
+{
+  public class CheckSolution : BuildTaskBase
+  {
+
+    private string _projectPath;
+    public CheckSolution(string projectPath="") : base("check-solution")
```
Comment (JakobChristensen) [inline src/Sitecore.Pathfinder.Console/Building/Checking/CheckSolution.cs]: Not too fond of default values in constructor parameters.

File: src/Sitecore.Pathfinder.Console/Building/Checking/CheckSolution.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
```
Comment (JakobChristensen) [inline src/Sitecore.Pathfinder.Console/Building/Checking/CheckSolution.cs]: I am unsure of the purpose of this task - it looks like a copy of the CheckProject task.

Is this a placeholder for coming work?

File: src/Sitecore.Pathfinder.Console/Building/Checking/CheckSolution.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    {
+      context.Trace.TraceInformation(Msg.C1041, Texts.Checking___);
+
+      TraceDiagnostics(context);
+    }
+
+    public override void WriteHelp(HelpWriter helpWriter)
+    {
+      helpWriter.Summary.Write("Checks the Solution for warnings and errors.");
+      helpWriter.Remarks.Write("SETTINGS *****TO BE DISCUSSED*****:");
```
Comment (JakobChristensen) [inline src/Sitecore.Pathfinder.Console/Building/Checking/CheckSolution.cs]: Let's discuss.

File: src/Sitecore.Pathfinder.Console/Building/Tasks/WriteJsonProjectDiagnostics.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            using (JsonWriter writer = new JsonTextWriter(sw))
+            {
+                serializer.Serialize(writer, solutionDiagnostic);
+            }
+        }
+
+        [NotNull]
+        protected virtual string GetOutputPath([NotNull]IBuildContext context)
+        {
+            string path = context.Configuration.GetString("op");
```
Comment (JakobChristensen) [inline src/Sitecore.Pathfinder.Console/Building/Tasks/WriteJsonProjectDiagnostics.cs]: "op" should be "write-json-project-diagnostics:output-path"

File: src/Sitecore.Pathfinder.Console/Building/Tasks/WriteJsonProjectDiagnostics.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            context.Trace.TraceInformation(Msg.C1041, Texts.Checking___);
+
+            WriteJsonDiagnostics(context);
+        }
+
+        public override void WriteHelp(HelpWriter helpWriter)
+        {
+            helpWriter.Summary.Write("Checks the project for warnings and errors.");
+            helpWriter.Remarks.Write("SETTINGS:" + Environment.NewLine);
```
Comment (JakobChristensen) [inline src/Sitecore.Pathfinder.Console/Building/Tasks/WriteJsonProjectDiagnostics.cs]: Use WriteLine instead.

File: src/Sitecore.Pathfinder.Console/Building/Tasks/WriteJsonProjectDiagnostics.cs [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        [NotNull]
+        protected virtual string GetOutputPath([NotNull]IBuildContext context)
+        {
+            string path = context.Configuration.GetString("op");
+            return path ?? string.Empty;
+        }
+    }
+
+    internal class SolutionDiagnostic
+    {
```
Comment (JakobChristensen) [inline src/Sitecore.Pathfinder.Console/Building/Tasks/WriteJsonProjectDiagnostics.cs]: One class per file.

And people are usually not happy about internal classes in a pluggable framework - make it public.

File: src/Sitecore.Pathfinder.Core/Projects/Project.cs [hunk 7]
```diff
@@ -83,6 +83,8 @@ private Project()
 
         public ICollection<ISourceFile> SourceFiles { get; } = new List<ISourceFile>();
 
+        public string Name => Options.ProjectDirectory.Split('\\').Last();
```
Comment (JakobChristensen) [inline src/Sitecore.Pathfinder.Core/Projects/Project.cs]: This should probably be "ProjectName". 
Should be cached.
Possibly the ProjectName could be overwritten in the scconfig.json to make the project name independent of which directory it is located in.

File: src/Sitecore.Pathfinder.Server/Sitecore.Pathfinder.Server.csproj [hunk 8]
```diff
@@ -83,30 +83,26 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="Sitecore.ContentSearch">
-      <HintPath>..\..\lib\Sitecore\Sitecore.ContentSearch.dll</HintPath>
```
Comment (JakobChristensen) [inline src/Sitecore.Pathfinder.Server/Sitecore.Pathfinder.Server.csproj]: This is not right - references should be in the /lib directory.

Comment (alinulms) [inline src/Sitecore.Pathfinder.Server/Sitecore.Pathfinder.Server.csproj]: Loads of issues i see :) 

I am at the hospital now. Will look at it later.

**Comment by JakobChristensen** [comment]
I will close this pull request for now.

**Pull request closed without merging**